telemetry_core/src/main/java/com/newrelic/telemetry/metrics/MetricBatchSenderBuilder.java

```diff
@@ -80,16 +82,41 @@ public MetricBatchSenderBuilder enableAuditLogging() {
     return this;
   }
 
+  /**
+   * Provide your New Relic Insights Insert API key
+   *
+   * @see <a
+   *     href="https://docs.newrelic.com/docs/apis/getting-started/intro-apis/understand-new-relic-api-keys#user-api-key">New
+   *     Relic API Keys</a>
+   */
   public MetricBatchSenderBuilder apiKey(String apiKey) {
     this.apiKey = apiKey;
     return this;
   }
 
+  /**
+   * Provide an implementation for HTTP POST. {@link #build()} will throw if an implementation is
+   * not provided or this method is not called.
+   */
   public MetricBatchSenderBuilder httpPoster(HttpPoster httpPoster) {
     this.httpPoster = httpPoster;
     return this;
   }
 
+  /**
+   * Provide additional user agent information. The product is required to be non-null and
+   * non-empty. The version is optional, although highly recommended.
+   */
+  public MetricBatchSenderBuilder additionalUserAgent(String product, String version) {
+    Utils.verifyNonNull(product, "Product cannot be null in the additional user-agent.");
+    if (version == null || version.isEmpty()) {
+      additionalUserAgent = product;
+    } else {
+      additionalUserAgent = product + "/" + version;
+    }
+    return this;
```

Comment:
I understand why you chose `additional`, but it can't be called multiple times. I might think that the additive nature of the name means it would keep appending. Maybe "secondUserAgent" or "secondaryUserAgent"?

Reply:
I like `secondaryUserAgent`. Will make it so.